lua/autorun/server/ji_dsserver.lua

```diff
@@ -73,6 +74,24 @@ function JID.CanTarget( ent )
     return true
 end
 
+function JID.CanBeUsed( ply, ent )
+    if not IsValid( ent ) then return false end
+    if not IsValid( ply ) then return false end
+
+    if CFCPvp then
+        local owner = ent:GetCreator()
+        if IsValid( owner ) and owner:IsPlayer() then
+            if owner:IsInBuild() and owner ~= ply then return false end
+            if owner:IsInPVP() and ply:IsInBuild() then return false end
+        end
+    end
+
+    local canBeUsed = hook.Run( "JIDCanBeUsed", ent )
+    if canBeUsed == false then return false end
+
+    return true
```

Comment:
It's starting to get to the point where extracting the pvp code to cfc_pvp would make sense

```suggestion
    local canBeUsed = hook.Run( "JIDCanBeUsed", ent )
    if canBeUsed == false then return false end

    if not CFCPvp then return true end

    local owner = ent:GetCreator()
    if IsValid( owner ) and owner:IsPlayer() then
        if owner:IsInBuild() and owner ~= ply then return false end
        if owner:IsInPVP() and ply:IsInBuild() then return false end
    end

    return true
```